dashboard/src/actions/catalog.ts

```diff
@@ -60,16 +62,44 @@ const actions = [
 
 export type ServiceCatalogAction = ActionType<typeof actions[number]>;
 
+function isEmptyDeep(obj: any): boolean {
```

Comment:
Please add a description of what this function does. Mainly because it seems to have a recursive behavior.